File: LC49.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        d = {}
        l = []

        for i in strs:
            s = i
            i = "".join(sorted(i))
            if i not in d:
                d[i] = [s]
            else:
                d[i].append(s)
        
        for i in d:
            l.append(d[i])
        
        return l

File: LC49.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector <vector <string>> res;
        unordered_map <string, vector<string>> m;

        for(auto i : strs){
            string s = i;
            sort(i.begin(), i.end());
            m[i].push_back(s);
        }

        for(auto i : m)
            res.push_back(i.second);

        return res;
    }
};
